### PR TITLE
Merge Juno's changes into upstream

### PR DESCRIPTION
**Pull request by faddat**

### CODE REVIEW

File: x/tokenfactory/client/cli/tx.go [hunk 1]
```diff
@@ -11,8 +13,11 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/CosmWasm/token-factory/x/tokenfactory/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
+var DeniedDenoms = [1]string{"juno"}
```
Comment (alpe) [inline x/tokenfactory/client/cli/tx.go:19]: Why "juno" here?

File: scripts/test_node.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
```
Comment (alpe) [inline scripts/test_node.sh:1]: Where is this script called?

File: scripts/test_metadata.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
+export KEY="tf1" # wasm1hj5fveer5cjtn4wd6wstzugjfdxzl0xpvsr89g
```
Comment (alpe) [inline scripts/test_metadata.sh:1]: Where is this script called?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# This workflow will build a golang project
```
Comment (alpe) [inline .github/workflows/build.yml:1]: The default build system is circleCI. Changing this, should be discussed in an issue first

File: proto/osmosis/tokenfactory/v1beta1/authorityMetadata.proto [hunk 5]
```diff
@@ -4,7 +4,7 @@ package osmosis.tokenfactory.v1beta1;
 import "gogoproto/gogo.proto";
 import "cosmos/base/v1beta1/coin.proto";
 
-option go_package = "github.com/osmosis-labs/osmosis/v12/x/tokenfactory/types";
+option go_package = "github.com/CosmWasm/token-factory/x/tokenfactory/types";
```
Comment (alpe) [inline proto/osmosis/tokenfactory/v1beta1/authorityMetadata.proto:7]: 👍

**Review by alpe — CHANGES_REQUESTED** [review]
Thanks for the PR! I appreciate your effort to keep this module updated.
To contribute to the project, it would help to have PRs with a smaller scope that addresses a single issue.

File: x/tokenfactory/client/cli/tx.go (hunk 1 above)
Comment (reecepbcups) [reply x/tokenfactory/client/cli/tx.go:19]: This is being removed in another PR from our fork. **Resolved**

File: scripts/test_node.sh (hunk 2 above)
Comment (reecepbcups) [reply scripts/test_node.sh:1]: This is just a testing script to ensure the chain starts up correctly. Purely for development, not CI

File: scripts/test_metadata.sh (hunk 3 above)
Comment (reecepbcups) [reply scripts/test_metadata.sh:1]: This is just a testing script to ensure the chain starts up correctly. Purely for development, not CI. I am removing this, only meant to be local oops

**Comment by reecepbcups** [comment]
Should hold off on merge until https://github.com/CosmosContracts/token-factory/pull/9 is in

**Comment by alpe** [comment]
Closing this in favour of #20

**Pull request closed without merging**